src/consensus/pbft/pbftmetadata.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "node/entities.h"
+#include "node/rpc/jsonrpc.h"
+
+#include <msgpack-c/msgpack.hpp>
+#include <vector>
+
+namespace pbft
+{
+  struct Request
```

Comment:
This is the same `struct` as `ccf_req` in `pbftconfig.h`. I think we should remove the `ccf_req` definition and use this instead. With regards to serialisation, it would be worth measuring the performance impact on the PBFT end-to-end test, using msgpack instead of manual serialisation (i.e. `serialise()` and `deserialise()` in `pbftconfig.h`)

Reply:
using `msgpack` instead of the manual `serialise()` and `deserialise()` affects performance enough to want to keep the manual methods around (smallbank goes down from ~ 3812.6tx/s to 2952.66tx/s) so I'll keep the manual methods for now.